knowledge-base/adding-web-report-designer-in-net-core-mvc.md

```diff
@@ -0,0 +1,217 @@
+            {
+                DefinitionStorage = new FileDefinitionStorage(
+                    Path.Combine(sp.GetService<Microsoft.AspNetCore.Hosting.IHostingEnvironment>().ContentRootPath, "Reports")),
+                SettingsStorage = new FileSettingsStorage(
+                    Path.Combine(sp.GetService<Microsoft.AspNetCore.Hosting.IHostingEnvironment>().ContentRootPath, "Reports")),
+            });
+        }
+
+        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+            else
+            {
+                app.UseExceptionHandler("/Home/Error");
+            }
+            app.UseStaticFiles();
+
+            app.UseRouting();
+
+            app.UseAuthorization();
+
+            app.UseEndpoints(endpoints =>
+            {
+                endpoints.MapControllers();
+                endpoints.MapControllerRoute(
+                    name: "default",
+                    pattern: "{controller=Home}/{action=Index}/{id?}");
+            });
+        }
+    }
+```
+
+6. The initialization of the designer has to look as follows:
+```Javascript
+<!DOCTYPE html>
+<html xmlns="http://www.w3.org/1999/xhtml">
+<head>
+    <script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
+    <link href="https://fonts.googleapis.com/css?family=Roboto:400,500&display=swap" rel="stylesheet">
+
+    <title>Telerik HTML5 Report Viewer Demo</title>
+
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+
+    <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
+
+</head>
+<body>
+
+    <div id="webReportDesigner" class="">
+        loading...
+    </div>
+
+</body>
+</html>
+
+@section scripts
+    {
+    <script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
+    <script src="https://kendo.cdn.telerik.com/2020.1.114/js/kendo.all.min.js"></script>
+
+    <script src="/api/reports/resources/js/telerikReportViewer"></script>
+
+    <script src="api/reportdesigner/designerresources/js/webReportDesigner-14.1.20.618.min.js/"></script>
+
+    <script type="text/javascript">
+        $(document).ready(function () {
+            $("#webReportDesigner").telerik_WebReportDesigner({
+                toolboxArea: {
+                    layout: "list" //Change to "grid" to display the contents of the Components area in a flow grid layout.
+                },
+                serviceUrl: "api/reportdesigner/",
+                report: "Barcodes Report.trdp"
+            }).data("telerik_WebDesigner");
+        });
+    </script>
+}
+
+```
+
+7. If you are using [SQL DataSource](../sqldatasource) with Shared connection string, add a connection string in appsettings.json file:
```

Comment:
You may add a link to the other possible syntaxes of the connection string declaration in the JSON configuration